org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/WorkspaceManager.java

```diff
@@ -182,6 +182,15 @@ protected List<WorkspaceFolder> getWorkspaceFolders() {
 		return workspaceFolders;
 	}
 	
+	/**
+	 * @return the open document map, never {@code null}
+	 * 
+	 * @since 2.28
+	 */
+	protected Map<URI, Document> getOpenDocuments() {
```

Comment:
You may want to extract a method `createOpenDocuments` that's used in the field initializer such that clients can provide a ConcurrentHashMap or a some other kind of map implementation according to their liking.